subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdExtension.java

```diff
@@ -34,33 +36,37 @@ public class PmdExtension extends CodeQualityExtension {
 
     private final Project project;
 
-    private List<String> ruleSets;
     private TargetJdk targetJdk;
     private TextResource ruleSetConfig;
     private ConfigurableFileCollection ruleSetFiles;
     private boolean consoleOutput;
+    private final ListProperty<String> ruleSets;
     private final Property<Integer> rulesMinimumPriority;
     private final Property<Integer> maxFailures;
     private final Property<Boolean> incrementalAnalysis;
     private final Property<Integer> threads;
 
     public PmdExtension(Project project) {
         this.project = project;
-        this.rulesMinimumPriority = project.getObjects().property(Integer.class).convention(5);
-        this.incrementalAnalysis = project.getObjects().property(Boolean.class).convention(true);
-        this.maxFailures = project.getObjects().property(Integer.class).convention(0);
-        this.threads = project.getObjects().property(Integer.class).convention(1);
+        this.rulesMinimumPriority = project.getObjects().property(Integer.class);
+        this.incrementalAnalysis = project.getObjects().property(Boolean.class);
+        this.maxFailures = project.getObjects().property(Integer.class);
+        this.threads = project.getObjects().property(Integer.class);
+        this.ruleSets = project.getObjects().listProperty(String.class);
     }
 
     /**
      * The built-in rule sets to be used. See the <a href="https://pmd.github.io/pmd-6.39.0/pmd_rules_java.html">official list</a> of built-in rule sets.
      *
+     * If not configured explicitly, the returned conventional value is "category/java/errorprone.xml", unless {@link #getRuleSetConfig()} returns
```

Comment:
```suggestion
     * If not configured explicitly, the returned conventional value is "category/java/errorprone.xml", unless {@link #getRuleSetConfig()} returns.
```

Missing dot at the end of the sentence.